src/Agent.Listener/SelfUpdater.cs

```diff
@@ -412,6 +412,21 @@ private async Task DownloadLatestAgent(CancellationToken token)
                 }
             }
 
+            if (!String.IsNullOrEmpty(AgentKnobs.DisableAuthenticodeValidation.GetValue(HostContext).AsString()))
+            {
+                Trace.Warning("Authenticode validation skipped for downloaded agent package since it is disabled currently by agent settings.");
+            }
+
+            var isValid = this.VerifyAgentAuthenticode(latestAgentDirectory);
```

Comment:
Since `Authenticode` code validation works only windows let's wrap it in the following if statement:
```suggestion
           if (PlatformUtil.RunningOnWindows)
           {
              var isValid = this.VerifyAgentAuthenticode(latestAgentDirectory);
           }
```